api/src/main/java/org/apache/iceberg/expressions/InclusiveMetricsEvaluator.java

```diff
@@ -408,6 +408,54 @@ public Boolean or(Boolean leftResult, Boolean rightResult) {
       return ROWS_MIGHT_MATCH;
     }
 
+    @Override
+    public <T> Boolean notStartsWith(BoundReference<T> ref, Literal<T> lit) {
+      Integer id = ref.fieldId();
+
+      // Iceberg does not implement SQL 3-boolean logic. Return ROWS_MIGHT_MATCH
+      // to allow the query engine to make its own decision regarding NULL values.
+      if (mayContainNull(id)) {
+        return ROWS_MIGHT_MATCH;
+      }
+
+      ByteBuffer prefixAsBytes = lit.toByteBuffer();
+
+      Comparator<ByteBuffer> comparator = Comparators.unsignedBytes();
+
+      // notStartsWith will match unless all values must start with the prefix. This happens when the lower and upper
+      // bounds both start with the prefix.
+      if (lowerBounds != null && upperBounds != null &&
+          lowerBounds.containsKey(id) && upperBounds.containsKey(id)) {
+        ByteBuffer lower = lowerBounds.get(id);
+        // if lower is shorter than the prefix, it can't start with the prefix
```

Comment:
I think let's try to keep pronouns out of this since I have difficulty parsing this statement. I'm always confused though.

// if lower is shorter than the prefix then lower doesn't start with the prefix

Reply:
I can update it to that. I didn't actually write the comments for this one. They came directly as a suggestion of somebody else. Though I can see how `it` is confusing as well.

Reply:
I updated both the `it` for lower and for upper in this file. Will check the other evaluators for usage of `it` as well.